diff -Naur armagetronad-0.2.8.2.1/src/network/md5.cpp armagetronad-0.2.8.2.1.oden/src/network/md5.cpp
--- armagetronad-0.2.8.2.1/src/network/md5.cpp	2004-11-29 11:26:42.000000000 +0100
+++ armagetronad-0.2.8.2.1.oden/src/network/md5.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -78,6 +78,7 @@
 
 #include <string>
 #include <memory>
+#include <cstring>
 
 #ifdef TEST
 /*
diff -Naur armagetronad-0.2.8.2.1/src/network/nKrawall.cpp armagetronad-0.2.8.2.1.oden/src/network/nKrawall.cpp
--- armagetronad-0.2.8.2.1/src/network/nKrawall.cpp	2005-07-13 23:04:21.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/network/nKrawall.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -44,6 +44,7 @@
 
 #include <stdlib.h>
 #include <string>
+#include <cstring>
 
 bool nKrawall::MayRequirePassword(tString& adress, unsigned int port)
 {
diff -Naur armagetronad-0.2.8.2.1/src/network/nNetObject.h armagetronad-0.2.8.2.1.oden/src/network/nNetObject.h
--- armagetronad-0.2.8.2.1/src/network/nNetObject.h	2006-01-15 00:04:33.000000000 +0100
+++ armagetronad-0.2.8.2.1.oden/src/network/nNetObject.h	2008-07-02 16:58:59.000000000 +0200
@@ -31,6 +31,7 @@
 #include "nNetwork.h"
 #include "tArray.h"
 #include "tConsole.h"
+#include <cstring>
 
 class nObserver;
 
diff -Naur armagetronad-0.2.8.2.1/src/network/nNetwork.cpp armagetronad-0.2.8.2.1.oden/src/network/nNetwork.cpp
--- armagetronad-0.2.8.2.1/src/network/nNetwork.cpp	2006-08-04 20:56:13.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/network/nNetwork.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -48,6 +48,7 @@
 #endif
 
 #include <deque>
+#include <cstring>
 
 // debug watchs
 #ifdef DEBUG
diff -Naur armagetronad-0.2.8.2.1/src/network/nSocket.cpp armagetronad-0.2.8.2.1.oden/src/network/nSocket.cpp
--- armagetronad-0.2.8.2.1/src/network/nSocket.cpp	2006-05-13 16:17:31.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/network/nSocket.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -44,6 +44,7 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <vector>
+#include <cstring>
 
 #ifndef WIN32
 #include <arpa/inet.h> 
diff -Naur armagetronad-0.2.8.2.1/src/render/rFont.cpp armagetronad-0.2.8.2.1.oden/src/render/rFont.cpp
--- armagetronad-0.2.8.2.1/src/render/rFont.cpp	2006-05-13 16:17:31.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/render/rFont.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -29,6 +29,7 @@
 #include "rScreen.h"
 #include "tConfiguration.h"
 #include <ctype.h>
+#include <cstring>
 
 #ifndef DEDICATED
 #include "rRender.h"
diff -Naur armagetronad-0.2.8.2.1/src/render/rModel.cpp armagetronad-0.2.8.2.1.oden/src/render/rModel.cpp
--- armagetronad-0.2.8.2.1/src/render/rModel.cpp	2005-11-18 00:38:17.000000000 +0100
+++ armagetronad-0.2.8.2.1.oden/src/render/rModel.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -28,6 +28,7 @@
 #include "rModel.h"
 #include <string>
 #include <fstream>
+#include <cstring>
 #include <stdlib.h>
 #include "rScreen.h"
 #include "tString.h"
diff -Naur armagetronad-0.2.8.2.1/src/render/rScreen.cpp armagetronad-0.2.8.2.1.oden/src/render/rScreen.cpp
--- armagetronad-0.2.8.2.1/src/render/rScreen.cpp	2005-12-05 21:30:46.000000000 +0100
+++ armagetronad-0.2.8.2.1.oden/src/render/rScreen.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -30,6 +30,7 @@
 #include "defs.h"
 
 #include <string>
+#include <cstring>
 #include "rTexture.h"
 #include "rScreen.h"
 #include "rSysdep.h"
diff -Naur armagetronad-0.2.8.2.1/src/tools/tArray.cpp armagetronad-0.2.8.2.1.oden/src/tools/tArray.cpp
--- armagetronad-0.2.8.2.1/src/tools/tArray.cpp	2005-06-15 19:09:09.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/tools/tArray.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -32,6 +32,7 @@
 #include <iostream>
 #include <stdlib.h>
 #include <string>
+#include <cstring>
 #include "tArray.h"
 
 void GrowingArrayBase::ComplainIfFull(){
diff -Naur armagetronad-0.2.8.2.1/src/tools/tCommandLine.cpp armagetronad-0.2.8.2.1.oden/src/tools/tCommandLine.cpp
--- armagetronad-0.2.8.2.1/src/tools/tCommandLine.cpp	2005-12-05 21:30:46.000000000 +0100
+++ armagetronad-0.2.8.2.1.oden/src/tools/tCommandLine.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -34,6 +34,7 @@
 #include    "tLocale.h"
 #include    "tConfiguration.h"
 #include    "tException.h"
+#include    <cstring>
 
 #ifdef WIN32
 #include    <windows.h>
diff -Naur armagetronad-0.2.8.2.1/src/tools/tConfiguration.cpp armagetronad-0.2.8.2.1.oden/src/tools/tConfiguration.cpp
--- armagetronad-0.2.8.2.1/src/tools/tConfiguration.cpp	2006-05-13 16:17:31.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/tools/tConfiguration.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -33,6 +33,7 @@
 #include <stdlib.h>
 #include <ctype.h>
 #include <string>
+#include <cstring>
 #include <sstream>
 #include "tString.h"
 #include "tToDo.h"
diff -Naur armagetronad-0.2.8.2.1/src/tools/tCrypt.cpp armagetronad-0.2.8.2.1.oden/src/tools/tCrypt.cpp
--- armagetronad-0.2.8.2.1/src/tools/tCrypt.cpp	2004-11-29 11:26:42.000000000 +0100
+++ armagetronad-0.2.8.2.1.oden/src/tools/tCrypt.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -29,6 +29,7 @@
 #include "tMemManager.h"
 
 #include <string>
+#include <cstring>
 
 class tCryptDummy: public tCrypt
 {
diff -Naur armagetronad-0.2.8.2.1/src/tools/tDirectories.cpp armagetronad-0.2.8.2.1.oden/src/tools/tDirectories.cpp
--- armagetronad-0.2.8.2.1/src/tools/tDirectories.cpp	2006-06-02 22:38:09.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/tools/tDirectories.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -32,6 +32,7 @@
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <cstring>
 
 #include "tLocale.h"
 #include "tDirectories.h"
diff -Naur armagetronad-0.2.8.2.1/src/tools/tError.cpp armagetronad-0.2.8.2.1.oden/src/tools/tError.cpp
--- armagetronad-0.2.8.2.1/src/tools/tError.cpp	2006-03-16 00:16:59.000000000 +0100
+++ armagetronad-0.2.8.2.1.oden/src/tools/tError.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -26,6 +26,7 @@
 */
 
 #include "config.h"
+#include <cstdlib>
 #include <iostream>
 #if HAVE_UNISTD_H
 #include <unistd.h>
diff -Naur armagetronad-0.2.8.2.1/src/tools/tLocale.cpp armagetronad-0.2.8.2.1.oden/src/tools/tLocale.cpp
--- armagetronad-0.2.8.2.1/src/tools/tLocale.cpp	2005-09-09 23:49:42.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/tools/tLocale.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -33,6 +33,7 @@
 
 #include <fstream>
 #include <string>
+#include <cstring>
 #include <map>
 
 class tLocaleSubItem; // identifies a single string in a single language
diff -Naur armagetronad-0.2.8.2.1/src/tools/tMemManager.cpp armagetronad-0.2.8.2.1.oden/src/tools/tMemManager.cpp
--- armagetronad-0.2.8.2.1/src/tools/tMemManager.cpp	2006-05-13 16:17:31.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/tools/tMemManager.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -33,6 +33,7 @@
 #include <iostream>
 #include <sstream>
 #include <stdio.h>  // need basic C IO since STL IO does memory management
+#include <cstring>
 #include "tMemManager.h"
 #include "tError.h"
 
diff -Naur armagetronad-0.2.8.2.1/src/tools/tRandom.cpp armagetronad-0.2.8.2.1.oden/src/tools/tRandom.cpp
--- armagetronad-0.2.8.2.1/src/tools/tRandom.cpp	2005-12-05 21:30:47.000000000 +0100
+++ armagetronad-0.2.8.2.1.oden/src/tools/tRandom.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -34,6 +34,8 @@
 #include    <stdlib.h>
 #endif
 
+#include <cstdlib>
+
 #include    "tError.h"
 #include    "tRecorder.h"
 
diff -Naur armagetronad-0.2.8.2.1/src/tools/tString.cpp armagetronad-0.2.8.2.1.oden/src/tools/tString.cpp
--- armagetronad-0.2.8.2.1/src/tools/tString.cpp	2006-05-13 16:17:31.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/tools/tString.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -31,6 +31,7 @@
 #include "tConfiguration.h"
 #include <ctype.h>
 #include <string>
+#include <cstring>
 #include <iostream>
 
 tString::tString(){
diff -Naur armagetronad-0.2.8.2.1/src/tron/gAIBase.cpp armagetronad-0.2.8.2.1.oden/src/tron/gAIBase.cpp
--- armagetronad-0.2.8.2.1/src/tron/gAIBase.cpp	2006-06-02 22:38:09.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/tron/gAIBase.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -43,6 +43,7 @@
 #include "tRandom.h"
 #include "tRecorder.h"
 #include <stdlib.h>
+#include <memory>
 
 #define AI_REACTION          0 
 #define AI_EMERGENCY         1 
diff -Naur armagetronad-0.2.8.2.1/src/tron/gCycle.cpp armagetronad-0.2.8.2.1.oden/src/tron/gCycle.cpp
--- armagetronad-0.2.8.2.1/src/tron/gCycle.cpp	2006-06-02 22:38:09.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/tron/gCycle.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -52,6 +52,7 @@
 #include "tMath.h"
 #include <stdlib.h>
 #include <fstream>
+#include <memory>
 
 #ifndef DEDICATED
 #define DONTDOIT
diff -Naur armagetronad-0.2.8.2.1/src/ui/uInput.cpp armagetronad-0.2.8.2.1.oden/src/ui/uInput.cpp
--- armagetronad-0.2.8.2.1/src/ui/uInput.cpp	2006-05-13 16:17:32.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/ui/uInput.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -33,6 +33,7 @@
 #include "rConsole.h"
 #include "uMenu.h"
 #include "tSysTime.h"
+#include <cstring>
 
 bool su_mouseGrab = false;
 
diff -Naur armagetronad-0.2.8.2.1/src/ui/uInputQueue.cpp armagetronad-0.2.8.2.1.oden/src/ui/uInputQueue.cpp
--- armagetronad-0.2.8.2.1/src/ui/uInputQueue.cpp	2006-05-13 16:17:32.000000000 +0200
+++ armagetronad-0.2.8.2.1.oden/src/ui/uInputQueue.cpp	2008-07-02 16:58:59.000000000 +0200
@@ -29,6 +29,7 @@
 #include "rScreen.h"
 #include "tConfiguration.h"
 #include <iostream>
+#include <cstring>
 
 #ifndef DEDICATED
 #include "rSDL.h"
